quixstreams/kafka/configuration.py

```diff
@@ -0,0 +1,147 @@
+        return self.as_printable_json()
+
+    bootstrap_servers: str
+    security_protocol: Annotated[
+        Optional[Literal["plaintext", "ssl", "sasl_plaintext", "sasl_ssl"]],
+        BeforeValidator(lambda v: v.lower() if v is not None else v),
+    ] = None
+
+    # ----------------- SASL SETTINGS -----------------
+    # Allows "sasl_mechanisms" (or "." if librdkafka) and sets it to sasl_mechanism
+    sasl_mechanism: Annotated[
+        Optional[
+            Literal["GSSAPI", "PLAIN", "SCRAM-SHA-256", "SCRAM-SHA-512", "OAUTHBEARER"]
+        ],
+        Field(validation_alias=MECHANISM_ALIAS),
+        BeforeValidator(lambda v: v.upper() if v is not None else v),
+    ] = None
+    sasl_username: Optional[str] = None
+    sasl_password: Optional[SecretStr] = None
+    sasl_kerberos_kinit_cmd: Optional[str] = None
+    sasl_kerberos_keytab: Optional[str] = None
+    sasl_kerberos_min_time_before_relogin: Optional[int] = None
+    sasl_kerberos_service_name: Optional[str] = None
+    sasl_kerberos_principal: Optional[str] = None
+    # for oauth_cb, see https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#pythonclient-configuration
+    oauth_cb: Optional[Callable[[str], Tuple[str, float]]] = None
+    sasl_oauthbearer_config: Optional[str] = None
+    enable_sasl_oauthbearer_unsecure_jwt: Optional[bool] = None
+    oauthbearer_token_refresh_cb: Optional[Callable] = None
+    sasl_oauthbearer_method: Annotated[
+        Optional[Literal["default", "oidc"]],
+        BeforeValidator(lambda v: v.lower() if v is not None else v),
+    ] = None
+    sasl_oauthbearer_client_id: Optional[str] = None
+    sasl_oauthbearer_client_secret: Optional[SecretStr] = None
+    sasl_oauthbearer_scope: Optional[str] = None
+    sasl_oauthbearer_extensions: Optional[str] = None
+    sasl_oauthbearer_token_endpoint_url: Optional[str] = None
+
+    # ----------------- SSL SETTINGS -----------------
+    ssl_cipher_suites: Optional[str] = None
+    ssl_curves_list: Optional[str] = None
+    ssl_sigalgs_list: Optional[str] = None
+    ssl_key_location: Optional[str] = None
+    ssl_key_password: Optional[SecretStr] = None
+    ssl_key_pem: Optional[str] = None
+    ssl_certificate_location: Optional[str] = None
+    ssl_certificate_pem: Optional[str] = None
+    ssl_ca_location: Optional[str] = None
+    ssl_ca_pem: Optional[str] = None
+    ssl_ca_certificate_stores: Optional[str] = None
+    ssl_crl_location: Optional[str] = None
+    ssl_keystore_location: Optional[str] = None
+    ssl_keystore_password: Optional[SecretStr] = None
+    ssl_providers: Optional[str] = None
+    ssl_engine_location: Optional[str] = None
+    ssl_engine_id: Optional[str] = None
+    enable_ssl_certificate_verification: Optional[bool] = None
+    ssl_endpoint_identification_algorithm: Annotated[
+        Optional[Literal["none", "https"]],
+        BeforeValidator(lambda v: v.lower() if v is not None else v),
+    ] = None
```

Comment:
What's the reason for putting the attributes after the methods in this class?
I know the list is big, but let's stick to the usual approach when the class attributes go before the methods